packages/runtime/runtime-utils/src/serializer.ts

```diff
@@ -3,15 +3,99 @@
  * Licensed under the MIT License.
  */
 
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+
 import {
     IFluidHandle,
     IFluidHandleContext,
     IFluidSerializer,
+    ISerializedHandle,
 } from "@fluidframework/core-interfaces";
 import { RemoteFluidObjectHandle } from "./remoteObjectHandle";
 import { generateHandleContextPath } from "./dataStoreHandleContextUtils";
 import { isSerializedHandle } from "./utils";
 
+/**
+ * Recursively applies the given 'replacer' function to all objects in the 'input' object
+ * graph, returning a new object graph that substitutes the output of replacer.
+ *
+ * Note that this function does not apply the 'replacer' function to non-object/non-array
+ * primitives like numbers, strings, etc.
+ */
+// Perf: Passing 'context' so that 'replacer' can be a vanilla function with no closure
+//       yields an ~1.3x improvement (node 12 x64).
+function recursivelyReplace<TContext>(
+    context: TContext,
+    input: any,
+    replacer: (context: TContext, input: any) => any,
+) {
```

Comment:
This is the 'recursivelyReplaceHandles()' method below, lifted into a top-level function and generalized to accept a 'replacer' callback.